backend/core/src/listings/listings-csv-exporter.service.ts

```diff
@@ -152,7 +152,9 @@ export class ListingsCsvExporterService {
         "Unit Types": listing.unitGroupSummary?.unitTypes
           .map((unitType) => formatBedroom[unitType])
           .join(", "),
-        "AMI Chart": listing.unitGroup?.amiLevels.map((level) => level.amiChart?.name).join(", "),
+        "AMI Chart": [
+          ...new Set(listing.unitGroup?.amiLevels.map((level) => level.amiChart?.name)),
```

Comment:
this for sure works, I think its not the most efficient approach though.

could we instead use a .reduce here and verify uniqueness in the accumulator ourselves then join that together?

Reply:
So I had actually gone back and forth on this decision here in development... I ended up deciding that using reduce() as a filter/map replacement sacrifices more readability than it would gain in efficiency (if I'm understanding correctly that it would have a linear runtime either way just one less iteration). The lack of readability is coming from my personal difficulty in reading reduce functions that aren't reducing an array to a single value. Much of this was inspired by this [video](https://www.youtube.com/watch?v=qaGjS7-qWzg) and basically summarized [here](https://dev.to/jasterix/is-reduce-bad-2hhd). Can also recognize that using set in this way may be less readable than a map + filter combo which I could also do and would definitely pull out into a helper. I'm totally good to change this, but just wanted to get your thoughts on these pieces first and see what you'd prefer @YazeedLoonat